h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java

```diff
@@ -1301,6 +1301,13 @@ private int run2(String[] args) throws Exception {
     if (principal != null && keytabPath != null) {
       UserGroupInformation.setConfiguration(conf);
       UserGroupInformation.loginUserFromKeytab(principal, keytabPath);
+      // performs user impersonation (will only work if core-site.xml has hadoop.proxyuser.*.* props set on name node
+      if (runAsUser != null) {
+        System.out.println("Attempting to securely impersonate user, " + runAsUser);
+        UserGroupInformation currentEffUser = UserGroupInformation.getLoginUser();
+        UserGroupInformation proxyUser = UserGroupInformation.createProxyUser(runAsUser, currentEffUser);
+        UserGroupInformation.setLoginUser(proxyUser);
```

Comment:
Curious question: Is this common practice? The static method in documentation/code is marked as private/unstable:

```java
@InterfaceAudience.Private
@InterfaceStability.Unstable
@VisibleForTesting
public synchronized static void setLoginUser(UserGroupInformation ugi) {
  // if this is to become stable, should probably logout the currently
  // logged in ugi if it's different
  loginUser = ugi;
}
```